File: problems/missingNumber/missing_number.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  int missingNumber(vector<int> &nums) {
    if (nums.empty())
      return 0;
    int length = nums.size();
    for (int i = 0; i < length;) {
      if (nums[i] < length and nums[i] != i) {
        swap(nums[i], nums[nums[i]]);
      } else {
        ++i;
      }
      for (int i = 0; i < length; ++i) {
        if (nums[i] != i) {
          return i;
        }
      }
      return length;
    }
  }
};

File: problems/missingNumber/missing_number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        if not nums:
            return 0

        length = len(nums)
        nums.sort()
        result = ""
        for i in range(0, length):
            if nums[i] != i:
                result = i
                break
        if result == "":
            return length
        return result
        